core/src/test/java/org/kiwix/kiwixmobile/core/dao/NewBookmarksDaoTest.kt

```diff
@@ -51,15 +53,69 @@ internal class NewBookmarksDaoTest {
   @Test
   fun `get bookmarks without current book`() {
     val zimFileReader: ZimFileReader = mockk()
+    val queryBuilder: QueryBuilder<BookmarkEntity> = mockk()
+    every { box.query() } returns queryBuilder
+    every { queryBuilder.order(BookmarkEntity_.bookmarkTitle) } returns queryBuilder
+    val query: Query<BookmarkEntity> = mockk(relaxed = true)
+    every { queryBuilder.build() } returns query
+    val id = 0L
+    val zimId = "zimId"
+    val zimName = "zimName"
+    val zimFilePath = "zimFilePath"
+    val bookmarkUrl = "bookmarkUrl"
+    val bookmarkTitle = "bookmarkTitle"
+    val favicon: String? = "favicon"
+    val bookmarkEntities: List<BookmarkEntity> = listOf(
+      bookmarkEntity(
+        id = id,
+        zimId = zimId,
+        zimName = zimName,
+        zimFilePath = zimFilePath,
+        bookmarkUrl = bookmarkUrl,
+        bookmarkTitle = bookmarkTitle,
+        favicon = favicon
+      )
+    )
+    every { query.find() } returns bookmarkEntities
+    val bookmarkItem: BookmarkItem = mockk()
+    every { bookmarkItem.databaseId } returns id
+    every { bookmarkItem.zimId } returns zimId
+    every { bookmarkItem.zimName } returns zimName
+    every { bookmarkItem.zimFilePath } returns zimFilePath
+    every { bookmarkItem.bookmarkUrl } returns bookmarkUrl
+    every { bookmarkItem.bookmarkTitle } returns bookmarkTitle
+    every { bookmarkItem.favicon } returns favicon
+    assertThat(
+      newBookmarksDao.getBookmarks(
+        false,
+        zimFileReader
+      )
+    ).containsExactlyInAnyOrderElementsOf(
+      listOf(
+        bookmarkItem(
+          databaseId = id,
+          zimId = zimId,
+          zimName = zimName,
+          zimFilePath = zimFilePath,
+          bookmarkUrl = bookmarkUrl,
+          bookmarkTitle = bookmarkTitle,
+          favicon = favicon
+        )
+      )
+    )
+  }
+
+  @Test
+  fun `get bookmarks without current book when zimFileReader is null`() {
```

Comment:
you only need 1 test without current book to check that execution path, this test should be "with current book"

to test this
```
 if (fromCurrentBook) {
      equal(BookmarkEntity_.zimName, zimFileReader?.name ?: "")
    }
```
we want 1 test `fromCurrentBook=true zimFileReader=null` and another test `fromCurrentBook=true zimFileReader=mockThatWillReturnASpecificName()`